### DISCUSSION TITLE
Ability use Hextra in multiple sub-directories

### DISCUSSION THREAD

**Discussion by fmunteanu** [imfing/hextra#401] [Q&A]
Hi @imfing,

I've struggling to find a proper solution for my problem, without much success. My goal is to use Hextra with the following design pattern:

- /docs - website frontpage
- /docs/projectA - website project A
- /docs /projectB - website project B

Each project will have different menus and must also a dedicated `hextra-home` page. To understand better what I mean, please visit the currently [deployed website](https://axivo.com). The only way I managed to achieve the expected results, was by building Hextra for each directory, which introduces files duplication into `/public` directory, icons, sitemap, etc.

See my repository [files structure](https://github.com/axivo/website), for more details:

```
.
├── docs -- WEBSITE FRONTPAGE
│   ├── content
│   ├── resources
│   │   └── _gen
│   └── static
│       └── images
├── global -- COMMON FILES, used on both frontpage and several to be added projects
│   ├── i18n
│   ├── layouts
│   │   ├── partials
│   │   └── shortcodes
│   └── static
│       └── images
├── k3s-cluster -- FIRST PROJECT
│   ├── content
│   │   ├── tutorials
│   │   └── wiki
│   ├── resources
│   │   └── _gen
│   └── static
│       └── images
└── public -- WEBSITE FRONTPAGE
    ├── categories
    ├── css
    │   └── compiled
    ├── images
    ├── js
    ├── k3s-cluster -- FIRST PROJECT
    │   ├── tutorials
    │   └── wiki
    ├── lib
    │   └── flexsearch
    └── tags
```

Related to `hextra-home` page into sub-directories, I could customize the `single.html` file, but that will block me to use it on sub-directories. So I need to be able to use the actual `hextra-home` template into each sub-directory.

Another issue I'm facing is the ability to have specific menus, per sub-directory. Check the [deployed website](https://axivo.com), you can see specific menus defined. In real like, with Hugo, all you have to do is set the module mounts into `/docs` and everything works. This solution does not applies to me, unfortunately.

I'm wondering if you have any suggestions, I'm trying to find a proper solution for this, for over a week.

**Comment by imfing** [comment]
To be honest, I don't have much experience with multi-directory setup. 
I would recommend ask this in [Hugo forum](https://discourse.gohugo.io) which is a better place for general Hugo setup questions.

**Reply by fmunteanu** [reply]
No worries, [I already did](https://discourse.gohugo.io/t/50018). I was hoping you might have some insight, since you know the theme well.

**Comment by fmunteanu** [comment]
@imfing I made some progress, see [this PR](https://github.com/axivo/website/pull/104/files) with the related fixes. If I create the `/layouts/docs/hextra-home.html`, the page will render, but for some reason, I cannot disable the sidebar with `exclude: true`. If you could provide some guidance with this issue, it will be much appreciated.

![image](https://github.com/imfing/hextra/assets/19806136/d06b3baf-58e7-4768-afeb-41248e82e7d4)

Also, the menus show correctly into sidebar, but they don't load into top navbar. I would expect to have the top bar title and menus loaded from [/k3s-cluster/hugo.yaml](https://github.com/axivo/website/blob/5b16076784e0d0f8a1b6b8988ee9333b37f654c0/k3s-cluster/hugo.yaml), but instead, they are loaded from [/docs/hugo.yaml](https://github.com/axivo/website/blob/5b16076784e0d0f8a1b6b8988ee9333b37f654c0/docs/hugo.yaml):

![image](https://github.com/imfing/hextra/assets/19806136/b6214a41-cb7a-47f5-946a-72177e06e196)

**Comment by fmunteanu** [comment]
@imfing looking at https://github.com/imfing/hextra/blob/main/layouts/hextra-home.html, I see you defined in there the following condition:

```
{{ partial "sidebar.html" (dict "context" . "disableSidebar" true) }}
```

I don't understand where the `context` dictionary is used/defined, so the above settings are taken into account.

**Comment by fmunteanu** [comment]
@imfing I would appreciate if you can take few minutes of your time to help me understand how I disable the sidebar. I don't understand how this is done for [hextra-home.html](https://github.com/imfing/hextra/blob/main/layouts/hextra-home.html).

**Reply by imfing** [reply]
you can just remove the line containing the `sidebar`

**Reply by fmunteanu** [reply]
@imfing I'm going to push a PR that fixes the issue the correct way, without removing anything from your templates. Also, you're hardcoding the `.Site.Menus.main` map, which takes away the ability to add other menus, as per Hugo documentation.

I'll add a condition from Hugo [documentation](https://gohugo.io/content-management/menus/) into `navbar.html`, something like:

```
{{- $menu := .Site.Params.sectionPagesMenu | default "main" }}
```

Also, you use `menu:` instead of `menus:` in your [deployment](https://github.com/imfing/hextra/blob/main/exampleSite/hugo.yaml#L55), see above documentation.

**Comment by fmunteanu** [comment]
@imfing I created #406.
